test/e2e/e2e_suite_test.go

```diff
@@ -1,48 +1,67 @@
 package e2e
 
 import (
+	"context"
 	"flag"
 	"fmt"
 	"testing"
 	"time"
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/rand"
 
+	clusterv1beta2 "open-cluster-management.io/api/cluster/v1beta2"
 	operatorapiv1 "open-cluster-management.io/api/operator/v1"
 )
 
 var t *Tester
 
 var (
-	clusterName           string
-	klusterletName        string
-	agentNamespace        string
-	hubKubeconfig         string
-	nilExecutorValidating bool
-	deployKlusterlet      bool
-	managedKubeconfig     string
-	eventuallyTimeout     time.Duration
-	registrationImage     string
-	workImage             string
-	singletonImage        string
+	// kubeconfigs
+	hubKubeconfig     string
+	managedKubeconfig string
+
+	// customized test parameters
 	klusterletDeployMode  string
+	nilExecutorValidating bool
+
+	// images
+	registrationImage string
+	workImage         string
+	singletonImage    string
+
+	eventuallyTimeout time.Duration
 )
 
 func init() {
-	flag.StringVar(&clusterName, "cluster-name", "", "The name of the managed cluster on which the testing will be run")
 	flag.StringVar(&hubKubeconfig, "hub-kubeconfig", "", "The kubeconfig of the hub cluster")
-	flag.BoolVar(&nilExecutorValidating, "nil-executor-validating", false, "Whether validate the nil executor or not (default false)")
-	flag.BoolVar(&deployKlusterlet, "deploy-klusterlet", false, "Whether deploy the klusterlet on the managed cluster or not (default false)")
 	flag.StringVar(&managedKubeconfig, "managed-kubeconfig", "", "The kubeconfig of the managed cluster")
-	flag.DurationVar(&eventuallyTimeout, "eventually-timeout", 60*time.Second, "The timeout of Gomega's Eventually (default 60 seconds)")
+
+	flag.StringVar(&klusterletDeployMode, "klusterlet-deploy-mode", string(operatorapiv1.InstallModeDefault), "The image of the work")
+	flag.BoolVar(&nilExecutorValidating, "nil-executor-validating", false, "Whether validate the nil executor or not (default false)")
+
 	flag.StringVar(&registrationImage, "registration-image", "", "The image of the registration")
 	flag.StringVar(&workImage, "work-image", "", "The image of the work")
 	flag.StringVar(&singletonImage, "singleton-image", "", "The image of the klusterlet agent")
-	flag.StringVar(&klusterletDeployMode, "klusterlet-deploy-mode", string(operatorapiv1.InstallModeDefault), "The image of the work")
+
+	flag.DurationVar(&eventuallyTimeout, "eventually-timeout", 60*time.Second, "The timeout of Gomega's Eventually (default 60 seconds)")
 }
 
+// The e2e will always create one universal klusterlet, the developers can reuse this klusterlet in their case
+// but also pay attention, because the klusterlet is shared, so the developers should not delete the klusterlet.
+// And there might be some side effects on other cases if the developers change the klusterlet's spec for their cases.
+var (
+	universalClusterName    string
```

Comment:
can we also define the default value here?

Reply:
Yes we can, updated.